api/src/groups/test_groups.py

```diff
@@ -5,993 +5,816 @@
+    assert resp.json['active'] == True
+    #WHEN test deactivate meeting
+    resp = auth_client.patch(url_for('groups.read_one_meeting', meeting_id=first_meeting.id), json= {'active': False})
+    assert resp.json['active'] == False
+
+#
 @pytest.mark.smoke
 def test_invalid_update_meeting(auth_client):
-    # GIVEN a database with a number of groups and meetings
-    count = random.randint(3, 11)
-    generate_managers(auth_client)
-    create_multiple_groups(auth_client.sqla, count)
-    create_multiple_meetings(auth_client.sqla, count)
-
-    valid_meeting = meeting_object_factory(auth_client.sqla)
-
-    # WHEN we try to update an invalid meeting
-    resp = auth_client.patch(url_for('groups.update_meeting', meeting_id=-1), json=valid_meeting)
-    # THEN we expect an error code
-    assert resp.status_code == 404
-
-    original_meeting = auth_client.sqla.query(Meeting).first()
-    modified_meeting = valid_meeting
-    modified_meeting['invalid_field'] = 4
-
-    # WHEN we try to update an existing meeting with invalid payload
-    resp = auth_client.patch(url_for('groups.update_meeting', meeting_id=original_meeting.id), json=modified_meeting)
-
-    # THEN we assume the incorrect status code
+    # GIVEN an empty database
+    # WHEN we update one meeting with invalid information
+    create_multiple_groups(auth_client.sqla,1)
+    create_multiple_meetings(auth_client.sqla, 3)
+    # WHEN we get one meeting
+    first_meeting = auth_client.sqla.query(Meeting).first()
+    resp = auth_client.patch(url_for('groups.read_one_meeting', meeting_id=first_meeting.id), json= {'description': 123})
     assert resp.status_code == 422
 
-
+@pytest.mark.smoke
 def test_delete_meeting(auth_client):
-    # GIVEN a database with meetings and attendances
-    count = random.randint(3, 11)
-    generate_managers(auth_client)
-    create_multiple_groups(auth_client.sqla, count)
-    create_multiple_members(auth_client.sqla, count)
-    create_multiple_meetings(auth_client.sqla, count)
-    create_attendance(auth_client.sqla, 1)
-    # WHEN we delete a meeting
-    meeting_id = auth_client.sqla.query(Meeting.id).first()[0]
-    resp = auth_client.delete(url_for('groups.delete_meeting', meeting_id=meeting_id))
-    # THEN we assume the correct status code
-    assert resp.status_code == 200
-    # THEN we should not have that meeting in the database
-    assert auth_client.sqla.query(Meeting).filter_by(id=meeting_id).count() == 0
-    # THEN we should not have any attendances related to that meeting in the attendance table
-    assert auth_client.sqla.query(Attendance).filter_by(meeting_id=meeting_id).count() == 0
-
-    # WHEN we delete that meeting again
-    resp = auth_client.delete(url_for('groups.delete_meeting', meeting_id=meeting_id))
-    # THEN we should get an error code
-    assert resp.status_code == 404
-
-
-def test_activate_meeting(auth_client):
```

Comment:
Why are these tests deleted? Did they move somewhere else? Are they no longer relevant?

Reply:
Since there are many changes made to the database related to the `Group` module, I figured that most of those tests would need to be rewritten as well, and therefore removed all of them at once. For example, `test_read_all_meetings_by_group` is no longer relevant since we no longer have a an endpoint called `read_all_meetings_by_group` now. But there are perhaps parts that are better modified than removed and rewritten, which was my fault...
The `read_all_meetings_by_group` functionality can now be achieved with adding a query string to the "read_all" endpoint. To test that the query strings work properly (e.g. `...?where=group_id:4`), @quiteqiang 
 implemented some of the tests in `test_subset_group`; but since we now have a general mechanism to parse the query string, do you think we should write some general tests that test on the query string functionality, or should we rewrite them for all entities?
@quiteqiang is working on splitting large test functions into smaller ones, and adding tests for other status codes, as well as adding some of the tests that are still relevant now (like `test_activate_meeting`).

Reply:
Regarding the tests for parsing queries, we should have comprehensive tests for that general functionality, but we _don't_ need to repeat those tests everywhere they're used.